packages/@aws-cdk/aws-ec2/lib/client-vpn-endpoint.ts

```diff
@@ -0,0 +1,407 @@
+   * @default - use mutual authentication
+   */
+  readonly userBasedAuthentication?: ClientVpnUserBasedAuthentication;
+
+  /**
+   * Whether to enable connections logging
+   *
+   * @default true
+   */
+  readonly logging?: boolean;
+
+  /**
+   * A CloudWatch Logs log group for connection logging
+   *
+   * @default - a new group is created
+   */
+  readonly logGroup?: logs.ILogGroup;
+
+  /**
+   * A CloudWatch Logs log stream for connection logging
+   *
+   * @default - a new stream is created
+   */
+  readonly logStream?: logs.ILogStream;
+
+  /**
+   * The AWS Lambda function used for connection authorization
+   *
+   * The name of the Lambda function must begin with the `AWSClientVPN-` prefix
+   *
+   * @default - no connection handler
+   */
+  readonly clientConnectionHandler?: IClientVpnConnectionHandler;
+
+  /**
+   * A brief description of the Client VPN endpoint.
+   *
+   * @default - no description
+   */
+  readonly description?: string;
+
+  /**
+   * The security groups to apply to the target network.
+   *
+   * @default - a new security group is created
+   */
+  readonly securityGroups?: ISecurityGroup[];
+
+  /**
+   * Specify whether to enable the self-service portal for the Client VPN endpoint.
+   *
+   * @default true
+   */
+  readonly selfServicePortal?: boolean;
+
+  /**
+   * The ARN of the server certificate
+   */
+  readonly serverCertificateArn: string;
+
+  /**
+   * Indicates whether split-tunnel is enabled on the AWS Client VPN endpoint.
+   *
+   * @see https://docs.aws.amazon.com/vpn/latest/clientvpn-admin/split-tunnel-vpn.html
+   *
+   * @default false
+   */
+  readonly splitTunnel?: boolean;
+
+  /**
+   * The transport protocol to be used by the VPN session.
+   *
+   * @default TransportProtocol.UDP
+   */
+  readonly transportProtocol?: TransportProtocol;
+
+  /**
+   * The port number to assign to the Client VPN endpoint for TCP and UDP
+   * traffic.
+   *
+   * @default VpnPort.HTTPS
+   */
+  readonly port?: VpnPort;
+
+  /**
+   * Information about the DNS servers to be used for DNS resolution.
+   *
+   * A Client VPN endpoint can have up to two DNS servers.
+   *
+   * @default - use the DNS address configured on the device
+   */
+  readonly dnsServers?: string[];
+
+  /**
+   * Subnets to associate to the client VPN endpoint.
+   *
+   * @default - the VPC default strategy
+   */
+  readonly vpcSubnets?: SubnetSelection;
+
+  /**
+   * Whether to authorize all users to the VPC CIDR
+   *
+   * This automatically creates an authorization rule. Set this to `false` and
+   * use `addAuthorizationRule()` to create your own rules instead.
+   *
+   * @default true
+   */
+  readonly authorizeAllUsersToVpcCidr?: boolean;
+}
+
+/**
+ * User-based authentication for a client VPN endpoint
+ */
+export abstract class ClientVpnUserBasedAuthentication {
+  /**
+   * Active Directory authentication
+   */
+  public static activeDirectory(directoryId: string): ClientVpnUserBasedAuthentication {
```

Comment:
Here again, maybe we can do `IClientVpnActiveDirectory` instead?

There are currently no L2s in `@aws-cdk/aws-directoryservice` but when they'll come there will be a dependency on `@aws-cdk/aws-ec2`

Reply:
We can always add a new factory method for that if and when it becomes necessary. Let's not worry about it too much for now.